app/controllers/medications_controller.rb

```diff
@@ -45,11 +45,24 @@ def edit
     redirect_to_medication(@medication)
   end
 
+  def add_refill(medicine)
+    if medicine.refill.nil?
```

Comment:
We should also test when `medicine.refill` is not nill!

What also happens when we update a medication - can we change the refill values?

Reply:
Updating medication works fine, no issues on that front :)

Reply:
So another important question to consider is whether we want add_refill to be called when updating a medication.

Reply:
Actually, we'll have to call add_refill for updating. It's possible that the user would empty out the refill field in the form. I can update this too!